src/llrt-function.ts

```diff
@@ -154,6 +163,29 @@ export class LlrtFunction extends NodejsFunction {
       },
     });
 
+    if (props.useLambdaLayer && !props.llrtBinaryPath) {
+      this.ensureLayer(binaryUrl, binaryName, version);
+    }
+
     (this.node.defaultChild as CfnResource).addPropertyOverride('Runtime', 'provided.al2023');
   }
-}
+
+  private ensureLayer(binaryUrl: string, binaryName: string, version: string) {
+    const id = `${binaryName.replace("lambda", "layer")}-${version}`;
+    
+    let layer = Stack.of(this).node.tryFindChild(id);
+
+    if (!layer) {
+      layer = new LayerVersion(Stack.of(this), id, {
+        code: Code.fromDockerBuild(__dirname, {
+          buildArgs: {
+            URL: binaryUrl,
+          },
+        }),
+        layerVersionName: id,
+      });
```

Comment:
should set compatibleArchitectures. we don't need layerVersionName as it will be auto-generated.